decidim-budgets/app/commands/decidim/budgets/admin/import_proposals_to_budgets.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Budgets
+    module Admin
+      # A command with all the business logic when an admin imports proposals from
+      # one component to budget component.
+      class ImportProposalsToBudgets < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        def initialize(form)
+          @form = form
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          return broadcast(:invalid) unless @form.valid?
+
+          broadcast(:ok, import_proposals)
+        end
+
+        private
+
+        attr_reader :form
+
+        def import_proposals
+          proposals.map do |original_proposal|
+            next if proposal_already_copied?(original_proposal, target_component)
+
+            origin_attributes = original_proposal.attributes.except(
+              "id",
+              "title",
+              "body",
+              "created_at",
+              "updated_at",
+              "state",
+              "answer",
+              "answered_at",
+              "decidim_component_id",
+              "reference",
+              "proposal_votes_count",
+              "proposal_notes_count",
+              "proposal_endorsements_count",
+              "address",
+              "latitude",
+              "longitude",
+              "coauthorships_count",
+              "published_at"
+            )
+
+            project = Decidim::Budgets::Project.new(origin_attributes)
```

Comment:
Since we're creating two different entities I'd rather list the attributes that you want to copy instead of listing the ones we don't want.

Reply:
I have been reviewing it and it is not necessary to set 'origin_attributes' as you said, there are different entities